packages/medusa/src/services/pricing.ts

```diff
@@ -455,14 +456,39 @@ class PricingService extends TransactionBaseService {
       },
       0
     )
-    const tax = Math.round(price * rate)
-    const total = price + tax
 
-    return {
+    let taxAmount
```

Comment:
I opted for the following, not setting the price_includes_tax if not applicable by featureflag but otherwise I agree (though I changed the ternary a bit) 😄 

```
   const includesTax =
      this.featureFlagRouter.isFeatureEnabled(
        TaxInclusivePricingFeatureFlag.key
      ) && shippingOption.includes_tax

    const taxAmount = calculatePriceTaxAmount({
      taxRate: rate,
      price,
      includesTax,
    })
    const totalInclTax = includesTax ? price : price + taxAmount

    const result: PricedShippingOption = {
      ...shippingOption,
      price_incl_tax: totalInclTax,
      tax_rates: shippingOptionRates,
      tax_amount: taxAmount,
    }

    if (
      this.featureFlagRouter.isFeatureEnabled(
        TaxInclusivePricingFeatureFlag.key
      )
    ) {
      result.price_includes_tax = shippingOption.includes_tax
    }
```

Reply:
Perfect 👍 i am good with that too, it was mainly the rest